richtextfx/src/main/java/org/fxmisc/richtext/model/Paragraph.java

```diff
@@ -90,23 +103,33 @@ public String substring(int from) {
             return p;
         }
 
-        StyledText<S> left = segments.get(segments.size() - 1);
-        StyledText<S> right = p.segments.get(0);
-        if(Objects.equals(left.getStyle(), right.getStyle())) {
-            StyledText<S> segment = left.append(right.getText());
-            List<StyledText<S>> segs = new ArrayList<>(segments.size() + p.segments.size() - 1);
+        Segment<S> left = segments.get(segments.size() - 1);
+        Segment<S> right = p.segments.get(0);
+        if(canJoin(left, right)) {
+            Segment<S> segment = left.append(right.getText());
+            List<Segment<S>> segs = new ArrayList<>(segments.size() + p.segments.size() - 1);
             segs.addAll(segments.subList(0, segments.size()-1));
             segs.add(segment);
             segs.addAll(p.segments.subList(1, p.segments.size()));
             return new Paragraph<>(paragraphStyle, segs);
         } else {
-            List<StyledText<S>> segs = new ArrayList<>(segments.size() + p.segments.size());
+            List<Segment<S>> segs = new ArrayList<>(segments.size() + p.segments.size());
             segs.addAll(segments);
             segs.addAll(p.segments);
             return new Paragraph<>(paragraphStyle, segs);
         }
     }
 
+    private boolean canJoin(Segment<S> left, Segment<S> right) {
```

Comment:
This doesn't seem to be extensible... What happens if we or developers in their code include additional features that can be joined together? Wouldn't it make more sense to have some sort of segment-joining strategy?

For example:
- Strategy 1: inherit the left side's style
 - 1 plain Text + 1 plain Text = 1 plain Text
 - 1 BulletPointText + 1 plain Text = 1 BulletPointText
 - 1 plain Text + 1 BulletPointText = 1 plain Text
- Strategy 2: inherit the right side's style
 - 1 Empty Text + 1 Rich Text = 1 Rich Text
- Strategy 3: keep styles/objects distinct
 - 1 bolded text + 1 italicized text = 1 bolded text + 1 italicized text 
 - 1 Text + 1 Image = 1 Text + 1 Image
 - 1 Image + 1 Text = 1 Image + 1 Text
 - 1 Table + 1 Text = 1 Table + 1 Text
 - 1 Table + 1 Table = 1 Table + 1 Table
- Strategy 4: keep as two distinct objects, but change the style of latter
 - 1 Table + 1 BulletPointText = 1 Table + 1 Text

Having written out these examples, I'm thinking that a bullet-point style should be implemented as a paragraph style, not a text style.